static/js/topic_list.js

```diff
@@ -159,7 +178,41 @@ export class TopicListWidget {
         return this.my_stream_id;
     }
 
+    update_topic_search_text() {
+        const input = this.parent_elem.find("#filter-topic-input");
+        if (input.length) {
+            this.topic_search_text = input.val();
+        } else {
+            this.topic_search_text = "";
+        }
+    }
+
+    update_topic_search_input() {
+        const input = this.parent_elem.find("#filter-topic-input");
+        if (input.length) {
+            // update search text.
+            input.val(this.topic_search_text);
+            input.trigger("focus");
+
+            // setup event handlers.
+            const rebuild_list = () => this.build();
+            function clear_topic_search(e) {
+                e.stopPropagation();
+                input.val("");
+                input.trigger("blur");
+                rebuild_list();
```

Comment:
After playing with this for a while, I got into a state where this button doesn't work.

I think structurally, it'd be better to write this as a global handler for `$(body).on("click", "clear_search_topic_button"...) and have that call an exported `topic_list.clear_topic_filter` function instead.

Reply:
Yeah, I shifted this click handler set up to `click_handlers.js` which calls an exported function from topic list. This should surely resolve the state(where the button was not working).
(But while debugging manually for the state you got into, I could not get into it but it should be solved by the refactor you suggested if it had existed.)

Reply:
This sounds reasonable.